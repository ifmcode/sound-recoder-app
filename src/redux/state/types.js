// Types to identify dispatch action
const RecordingReducerType = {
  GET_RECORDINGS_SUCCESS: 'GET_RECORDINGS_SUCCESS',
  CREATE_RECORDING_SUCCESS: 'CREATE_RECORDING_SUCCESS',
  UPDATE_RECORDING_SUCCESS: 'UPDATE_RECORDING_SUCCESS',
  REMOVE_RECORDING_SUCCESS: 'REMOVE_RECORDING_SUCCESS',
  RECORDING_OPERATION_FAILED: 'RECORDING_OPERATION_FAILED',
}

// Store state status
const stateStatus = {
  ZERO: 'zero',
  SUCCESS: 'success',
  FAILED: 'failed',
};

export {
  RecordingReducerType,
  stateStatus,
}
