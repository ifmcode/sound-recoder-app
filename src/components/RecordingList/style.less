@import (reference) "../../styles/colors.less";
@import (reference) "../../styles/sizes.less";

#recording-list-container {
  padding: 10px;
  height: 200px;
  background-color: @black;
  border-radius: 8px;
  overflow: auto;
  #recording-list-wrapper {
    padding: 0;
    margin: 0;
    height: 100%;
    .recording-list-item {
      position: relative;
      color: @lightgrey;
      padding: 5px 2px 5px 30px;
      list-style-type: none;
      font-size: @font-size_18;
      text-transform: capitalize;
      cursor: pointer;

      .icon {
        position: absolute;
        left: 12px;
        width: 18px;
        height: 18px;
      }

      &.selected-recording {
        color: @white;
        background-color: rgba(255,255,255,0.3);
        border-radius: 5px;
      }
    }
  }

  .default-phrase-recording-list {
    color: grey;
    font-size: @font-size_18;
    text-align: center;
  }
}