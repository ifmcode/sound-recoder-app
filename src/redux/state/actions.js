import {RecordingReducerType} from './types';
import {getSavedRecordings, saveNewRecording, updateRecording, removeRecording}
  from '../../firebase/firebaseOperations';

/*
  Actions are used by the front to save or restore data from store
*/

const getRecordingsAction = () => async (dispatch) => {
  try {
    const recordings = await getSavedRecordings();
    dispatch({
      type: RecordingReducerType.GET_RECORDINGS_SUCCESS,
      payload: {
        recordings,
      }
    });
  } catch (error) {
    dispatchError(dispatch, error);
  }
}

const createRecordingAction = (newRecording) => async (dispatch) => {
  try {
    await saveNewRecording(newRecording);
    dispatch({
      type: RecordingReducerType.CREATE_RECORDING_SUCCESS,
      payload: {
        newRecording,
      }
    });
  } catch (error) {
    dispatchError(dispatch, error);
  }
}

const updateRecordingAction = (recordingUpdated) => async (dispatch) => {
  try {
    await updateRecording(recordingUpdated);
    dispatch({
      type: RecordingReducerType.UPDATE_RECORDING_SUCCESS,
      payload: {
        recordingUpdated,
      }
    });
  } catch (error) {
    dispatchError(dispatch, error);
  }
}

const removeRecordingAction = (recordingId) => async (dispatch) => {
  try {
    await removeRecording(recordingId);
    dispatch({
      type: RecordingReducerType.REMOVE_RECORDING_SUCCESS,
      payload: {
        recordingId,
      }
    });
  } catch (error) {
    dispatchError(dispatch, error);
  }
}

const dispatchError = (dispatch, error) => {
  dispatch({
    type: RecordingReducerType.RECORDING_OPERATION_FAILED,
    payload: {
      error: error,
    }
  });
}

export {
  getRecordingsAction,
  createRecordingAction,
  updateRecordingAction,
  removeRecordingAction,
}