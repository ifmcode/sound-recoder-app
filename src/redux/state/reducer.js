import {RecordingReducerType, stateStatus} from './types';

const initialData = {
  recordings: [],
  status: stateStatus.ZERO,
  error: null,
}

// Map that identify which function it calls when reducer receive a dispatch action
const ACTION_TYPE_FUNCTION_MAP = {
  [RecordingReducerType.GET_RECORDINGS_SUCCESS]: onGetRecordingsSuccess,
  [RecordingReducerType.CREATE_RECORDING_SUCCESS]: onCreateRecordingSuccess,
  [RecordingReducerType.UPDATE_RECORDING_SUCCESS]: onUpdateRecordingSuccess,
  [RecordingReducerType.REMOVE_RECORDING_SUCCESS]: onRemoveRecordingSuccess,
  [RecordingReducerType.RECORDING_OPERATION_FAILED]: onRecordingOperationFailed,
}

function onGetRecordingsSuccess(state, action) {
  return {
    ...state,
    recordings: action.payload.recordings,
    status: stateStatus.SUCCESS,
  };
}

function onCreateRecordingSuccess(state, action) {
  return {
    ...state,
    recordings: [...state.recordings, action.payload.newRecording],
    status: stateStatus.SUCCESS,
  };
}

function onUpdateRecordingSuccess(state, action) {
  const recordings = state.recordings,
    recordingUpdated = action.payload.recordingUpdated,
    indexOfRecordingtoUpdate = recordings.findIndex((recording) => recording.id === recordingUpdated.id);
  recordings[indexOfRecordingtoUpdate] = recordingUpdated;
  return {
    ...state,
    recordings,
    status: stateStatus.SUCCESS,
  };
}

function onRemoveRecordingSuccess(state, action) {
  const currentRecordings = state.recordings,
    recordingIdToDelete = action.payload.recordingId,
    recordings = currentRecordings.filter((recording) => recording.id !== recordingIdToDelete);
  return {
    ...state,
    recordings,
    status: stateStatus.SUCCESS,
  };
}

function onRecordingOperationFailed(state, action) {
  return {
    ...state,
    error: action.payload.error,
    status: stateStatus.FAILED,
  }
}

function soundRecordingReducer(state = initialData, action) {
  const fn = ACTION_TYPE_FUNCTION_MAP[action.type];
  return (fn)
    ? fn(state, action)
    : state;
}

export {
  soundRecordingReducer,
}
