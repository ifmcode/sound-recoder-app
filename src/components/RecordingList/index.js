import './style.less'
import React from 'react';
import Icon from '../Icon';
import {stateStatus} from '../../redux/state/types';

/**
 * Recording list component that show a default phrase or recordings name displayed as a list
 * @param {*} recordings 
 * @param {*} selectedRecording // This recording has a special style to show which recordint is selected
 * @param {*} storeStatus 
 * @param {*} onSelectRecordingHandler //Funtion that it call when a recording is clicked
 * @returns 
 */
const RecordingList = ({recordings, selectedRecording, storeStatus, onSelectRecordingHandler}) => {

  const getDefaultPhrase = () => {
    // Error case
    let defaultPhrase = <p className='default-phrase-recording-list'>- An error retrieving data has ocurred -</p>
    if (storeStatus === stateStatus.ZERO) {
      defaultPhrase = <p className='default-phrase-recording-list'>- Loading recordings -</p>
    } else if (storeStatus === stateStatus.SUCCESS) {
      defaultPhrase = <p className='default-phrase-recording-list'>- There are not recordings -</p>
    }
    return defaultPhrase;
  }

  const renderRecordingList = () => {
    let ret = getDefaultPhrase(); 
    if (recordings && recordings.length && selectedRecording) {
      ret = <ul id='recording-list-wrapper'>
        {
          recordings.map((recording) => {
            const isSelected = selectedRecording.id === recording.id;
            return <li
              key={recording.id}
              onClick={() => onSelectRecordingHandler(recording)}
              className={ isSelected
                ? 'recording-list-item selected-recording'
                : 'recording-list-item'}>
                {
                  isSelected &&
                    <Icon iconName="indicator"/>
                }
                {recording.name}
            </li>
          })
        }
      </ul>
    }
    return ret;
  }
  
  return (
    <div id="recording-list-container">
      {renderRecordingList()}
    </div>
  );
};

export default RecordingList;
