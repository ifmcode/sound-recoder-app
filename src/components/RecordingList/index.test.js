import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react'
import {stateStatus} from '../../redux/state/types';
import RecordingList from ".";

/**
 * Following the Object mother pattern we have this small fn that generates a valid object
 * that matches the structure of a recording
 *
 * @param {{ idNumber?: number, name?: string }} options
 * @returns {{ stream: string, name: string, id: string }}
 */
function getMockRecording({
  idNumber = Math.floor(Math.random() * 100),
  name = new Date().toISOString().split('.')[0].split('T').join(' ') + idNumber
}) {
  return {
    stream: "audioUrl",
    name,
    id: `id${idNumber}`
  }
}

/**
 * Applying the same pattern above we wrap the previous recording generator
 * to create a random list of recordings
 *
 * @param {number} length
 * @returns {{ stream: string, name: string, id: string }[]}
 */
function createMockRecordingList(length = 10) {
  const emptyList = new Array(length).fill(null);
  return emptyList.map(() => getMockRecording({}))
}

describe('Testing RecordingList component when store status is zero', () => {

  let wrapper;

  beforeEach(() => {
    const onSelectRecordingHandler = jest.fn();

    wrapper = render(
      <RecordingList
        recordings= {[]}
        selectedRecording= {null}
        storeStatus= {stateStatus.ZERO}
        onSelectRecordingHandler= {onSelectRecordingHandler}
      />
    );
  });
  
  test("should render the RecordingList component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without crashing showing default loading message', () => {
    expect(wrapper.container.querySelector(".default-phrase-recording-list")).toBeInTheDocument();
    expect(screen.getByText("- Loading recordings -")).toBeInTheDocument();
  });
})

describe('Testing RecordingList component with failed store status', () => {

  let wrapper;

  beforeEach(() => {
    const onSelectRecordingHandler = jest.fn();

    wrapper = render(
      <RecordingList
        recordings= {[]}
        selectedRecording= {null}
        storeStatus= {stateStatus.FAILED}
        onSelectRecordingHandler= {onSelectRecordingHandler}
      />
    );
  });
  
  test("should render the RecordingList component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without crashing showing error message', () => {
    expect(wrapper.container.querySelector(".default-phrase-recording-list")).toBeInTheDocument();
    expect(screen.getByText("- An error retrieving data has ocurred -")).toBeInTheDocument();
  });
})

describe('Testing RecordingList component when store status is success without recordings', () => {

  let wrapper;

  beforeEach(() => {
    const onSelectRecordingHandler = jest.fn();

    wrapper = render(
      <RecordingList
        recordings= {[]}
        selectedRecording= {null}
        storeStatus= {stateStatus.SUCCESS}
        onSelectRecordingHandler= {onSelectRecordingHandler}
      />
    );
  });
  
  test("should render the RecordingList component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without crashing the default message', () => {
    expect(wrapper.container.querySelector(".default-phrase-recording-list")).toBeInTheDocument();
    expect(screen.getByText("- There are not recordings -")).toBeInTheDocument();
  });
})

describe('Testing RecordingList component when store status is success with recordings', () => {

  let wrapper;
  const recordings = createMockRecordingList(),
    selectedRecording = recordings[0];

  beforeEach(() => {
    const onSelectRecordingHandler = jest.fn();

    wrapper = render(
      <RecordingList
        recordings= {recordings}
        selectedRecording= {selectedRecording}
        storeStatus= {stateStatus.SUCCESS}
        onSelectRecordingHandler= {onSelectRecordingHandler}
      />
    );
  });
  
  test("should render the RecordingList component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders the recording list instead the default phrase', () => {
    expect(screen.queryByText(selectedRecording.name)).toBeInTheDocument();
    expect(wrapper.container.querySelector(".default-phrase-recording-list")).toBeNull();
  });

  test('renders the recording list without crashing', () => {
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.queryAllByRole("listitem")).toHaveLength(10);
  });

  test('renders the selected recording with indicator icon', () => {
    expect(wrapper.container.querySelector(".indicator-icon")).toBeInTheDocument();
    expect(wrapper.container.querySelector(".selected-recording")).toBeInTheDocument();
  });
})