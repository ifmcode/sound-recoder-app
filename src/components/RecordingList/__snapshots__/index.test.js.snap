// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing RecordingList component when store status is success with recordings should render the RecordingList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="recording-list-container"
      >
        <ul
          id="recording-list-wrapper"
        >
          <li
            class="recording-list-item selected-recording"
          >
            <div
              class="icon indicator-icon"
            >
              <svg>
                indicator.svg
              </svg>
            </div>
            2022-01-26 14:24:0392
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0363
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0313
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0353
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0382
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0356
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0366
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0394
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0389
          </li>
          <li
            class="recording-list-item"
          >
            2022-01-26 14:24:0342
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="recording-list-container"
    >
      <ul
        id="recording-list-wrapper"
      >
        <li
          class="recording-list-item selected-recording"
        >
          <div
            class="icon indicator-icon"
          >
            <svg>
              indicator.svg
            </svg>
          </div>
          2022-01-26 14:24:0392
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0363
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0313
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0353
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0382
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0356
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0366
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0394
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0389
        </li>
        <li
          class="recording-list-item"
        >
          2022-01-26 14:24:0342
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing RecordingList component when store status is success without recordings should render the RecordingList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="recording-list-container"
      >
        <p
          class="default-phrase-recording-list"
        >
          - There are not recordings -
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="recording-list-container"
    >
      <p
        class="default-phrase-recording-list"
      >
        - There are not recordings -
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing RecordingList component when store status is zero should render the RecordingList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="recording-list-container"
      >
        <p
          class="default-phrase-recording-list"
        >
          - Loading recordings -
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="recording-list-container"
    >
      <p
        class="default-phrase-recording-list"
      >
        - Loading recordings -
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing RecordingList component with failed store status should render the RecordingList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="recording-list-container"
      >
        <p
          class="default-phrase-recording-list"
        >
          - An error retrieving data has ocurred -
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="recording-list-container"
    >
      <p
        class="default-phrase-recording-list"
      >
        - An error retrieving data has ocurred -
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
